{Release} Upgrade to Azure CLI 2.60.0 (#28817)
Azure/azure-cli: src/azure-cli/HISTORY.rst

Release History
===============

2.60.0
++++++

**ACR**

* [BREAKING CHANGE] `az acr connected-registry create`: Mode default value change from ReadWrite to ReadOnly (#28807)
* `az acr connected-registry create`: If data-endpoint disabled ask for confirmation to enable it instead of throwing an error (#28807)

**AKS**

* `az aks create/update`: Prompt warning during disablement about CR deletion (#28655)
* `az aks create/update`: Udpate RP registration code to work on azure monitor subscription (#28607)
* `az aks create/update`: Update to add default region for workspace creation in air gapped cloud (#28607)

> * Add option `--disable-windows-outbound-nat` for `az aks nodepool add` to add a Windows agent pool which the Windows OutboundNAT is disabled (#28806)

Reviewer: ```suggestion
* `az aks nodepool add`: Add parameter `--disable-windows-outbound-nat` to add a Windows agent pool which the Windows OutboundNAT is disabled (#28806)
```

> * Add option `--disable-windows-outbound-nat` for `az aks nodepool add` to add a Windows agent pool which the Windows OutboundNAT is disabled (#28806)

Reviewer 2: ```suggestion
* `az aks nodepool add`: Add `--disable-windows-outbound-nat` to add a Windows agent pool which the Windows OutboundNAT is disabled (#28806)
```

**App Service**

* `az webapp config container set`: Remove docker prefix and rename container related parameters (#28621)

> * `az webapp create` and `az webapp deployment slot create`: Rename `--docker-registry-server-user` and `--docker-registry-server-password` to `--container-registry-user` and `--container-registry-password` and add parameters `--container-image-name` and `container-registry-url` to deprecate `--deployment-container-image-name` (#28621)

Reviewer 2: ```suggestion
* `az webapp create` and `az webapp deployment slot create`: Rename `--docker-registry-server-user` and `--docker-registry-server-password` to `--container-registry-user` and `--container-registry-password` (#28621)
* `az webapp create` and `az webapp deployment slot create`: Add parameters `--container-image-name` and `container-registry-url` to deprecate `--deployment-container-image-name` (#28621)
```

* `az functionapp create`: Will validate that the provided runtime is supported by flex when creating a flex function app (#28795)
* `az functionapp list-flexconsumption-runtimes`: Add support for this new command so that it provides the list of supported flex runtimes when provided the `--location` and `--runtime` (#28795)

**ARM**

* `az bicep install`: Support additional process architectures with Bicep CLI (#28580)
* `az deployment`: Return better message on incorrect bicepparam file path (#28654)
* `az bicep format`: Replace `--newline` with `--newline-kind` (#28728)
* `az bicep publish`: Replace `--documentationUri` with `--documentation-uri` (#28728)

**Backup**

* `az backup vault update`: Take `--tenant-id` as input for resolving cross-tenant resource guard scenarios. Vault Immutability feature also has resource guard protection now, and support for the same has been added. (#28768)
* `az backup protection disable`: Take `--tenant-id` as input for resolving cross-tenant resource guard scenarios. Stop protection feature also has resource guard protection now, and support for the same has been added. (#28768)

**Batch**

> * `az batch pool create` Add `--upgrade-policy-mode` argument to support automatic OS upgrade (#28784)
> * `az batch pool create` Add `--enable-auto-os-upgrade` argument to enable automatic OS upgrade (#28784)
> * `az batch pool create` Add `--disable-auto-rollback` argument to disable OS image rollback feature (#28784)
> * `az batch pool create` Add `--defer-os-rolling-upgrade` argument to defer OS upgrades on the TVMs if they are running tasks (#28784)
> * `az batch pool create` Add `--use-rolling-upgrade-policy` argument to support OS rolling upgrade policy (#28784)
> * `az batch pool create` Add `--enable-cross-zone-upgrade` argument to support cross zone OS upgrade (#28784)
> * `az batch pool create` Add `--max-batch-instance-percent` argument to set the maximum percent of total VMs that will be upgraded in one batch (#28784)
> * `az batch pool create` Add `--max-unhealthy-instance-percent` argument to set the maximum percentage of the total VMs can be simultaneously unhealthy (#28784)
> * `az batch pool create` Add `--max-unhealthy-upgraded-instance-percent` argument to set the maximum percentage of upgraded VMs that can be found to be in an unhealthy state (#28784)
> * `az batch pool create` Add `--pause-time-between-batches` argument to set the wait time between batches in rolling OS upgrade (#28784)
> * `az batch pool create` Add `--prioritize-unhealthy-instances` argument to support upgrade all unhealthy VMs first (#28784)
> * `az batch pool create` Add `--rollback-failed-instances-on-policy-breach` argument to enable rollback failed instances to previous model if the Rolling Upgrade policy is violated (#28784)

Reviewer: ```suggestion
* `az batch pool create`: Add `--upgrade-policy-mode` argument to support automatic OS upgrade (#28784)
* `az batch pool create`: Add `--enable-auto-os-upgrade` argument to enable automatic OS upgrade (#28784)
* `az batch pool create`: Add `--disable-auto-rollback` argument to disable OS image rollback feature (#28784)
* `az batch pool create`: Add `--defer-os-rolling-upgrade` argument to defer OS upgrades on the TVMs if they are running tasks (#28784)
* `az batch pool create`: Add `--use-rolling-upgrade-policy` argument to support OS rolling upgrade policy (#28784)
* `az batch pool create`: Add `--enable-cross-zone-upgrade` argument to support cross zone OS upgrade (#28784)
* `az batch pool create`: Add `--max-batch-instance-percent` argument to set the maximum percent of total VMs that will be upgraded in one batch (#28784)
* `az batch pool create`: Add `--max-unhealthy-instance-percent` argument to set the maximum percentage of the total VMs can be simultaneously unhealthy (#28784)
* `az batch pool create`: Add `--max-unhealthy-upgraded-instance-percent` argument to set the maximum percentage of upgraded VMs that can be found to be in an unhealthy state (#28784)
* `az batch pool create`: Add `--pause-time-between-batches` argument to set the wait time between batches in rolling OS upgrade (#28784)
* `az batch pool create`: Add `--prioritize-unhealthy-instances` argument to support upgrade all unhealthy VMs first (#28784)
* `az batch pool create`: Add `--rollback-failed-instances-on-policy-breach` argument to enable rollback failed instances to previous model if the Rolling Upgrade policy is violated (#28784)
```

> **CI**
> 
> * Limit workflow target branches (#28706)
> 

Reviewer: ```suggestion
```

**Compute**

* `az vmss create`: Add support of configure the rolling mode upgrade policy during VMSS creation (#28761)
* `az vmss update`: Add new parameter `--max-surge` to support updating rolling upgrade policy max surge (#28761)
* `az capacity reservation group list`: Add new parameter `--resource-ids-only` to support retrieving the capacity reservation group resource ids (#28773)
* `az capacity reservation group create`: Change `--sharing profile` to support unsharing subscriptions by passing nothing (#28773)

**Containerapp**

* `az containerapp env create`: Support `--dapr-connection-string` to set application insights connection string used by Dapr to export service to service communication telemetry (#28625)
* Fix #28553: `az containerapp exec`: Fix the error of `inappropriate ioctl for device` (#28759)

**Monitor**

* `az monitor log-analytics workspace update`: Add parameter `--sku-name` (#28411)

**NetAppFiles**

> * `az netappfiles`: Update to api-version 2023-07-01 (#28575)

Reviewer 3: ```suggestion
```

> * `az netappfiles volume-group create`: Add `zones` argument to set Availability Zone for volume group volumes (#28709)

Reviewer: ```suggestion
* `az netappfiles volume-group create`: Add `--zones` argument to set Availability Zone for volume group volumes (#28709)
```

> * `az netappfiles volume create`: Update maximum value for `--usage-threshold` to support large volumes (#28709)
> * `az netappfiles volume update`: Update maximum value for `--usage-threshold` to support large volumes (#28709)

Reviewer 4: ```suggestion
* `az netappfiles volume create/update`: Update maximum value for `--usage-threshold` to support large volumes (#28709)
```

**Network**

* `az network virtual-appliance inbound-security-rule`: Support for Permanent Inbound Security Rule (#28697)

**RDBMS**

> * `az postgres flexible-server upgrade`: Add capability to perform major version upgrade to PG16. (#28687)

Reviewer 5: ```suggestion
* `az postgres flexible-server upgrade`: Add capability to perform major version upgrade to PG16 (#28687)
```
